dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMain.java

```diff
@@ -82,6 +86,20 @@ public static void main(String[] arguments)
 
     System.out.println(SymbolProblem.formatSymbolProblems(symbolProblems));
 
+    if (classPathResult != null && !symbolProblems.isEmpty()) {
+      ImmutableSet.Builder<Path> problematicJars = ImmutableSet.builder();
```

Comment:
You don't need this set, just print each jar as you find it. That should also make the order more predictable.

Reply:
That would explode the number of lines in the output. We need this de-duplication logic. When there's 100 linkage errors on 100 classes from a JAR file, this should not print the dependency path to the that JAR file 100 times.

Reply:
OK, that makes sense. Does the order of output matter though?

Reply:
No, the order does not matter. 

(Guava’s ImmutableSet guarantees a deterministic iteration order, which is not affected by JVM implementation.)